### PR TITLE
Custom type for JSON data

### PR DESCRIPTION
**Pull request by julienduchesne**
Currently, arbitrary data is being unmarshalled to an interface{}. 
By having a custom type, it makes it clear that the data is a map without having to cast. 
Also, we can add an unmarshal override so that numbers are unmarshalled to the native `float64` rather than the unexpected `json.Number`

### CODE REVIEW

File: pkg/custom_models/json.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	"encoding/json"
+)
+
+type JSON map[string]interface{}
+
+// Basic Unmarshal. Do not UseNumber (unmarshal to float64 over json.Number)
+func (j *JSON) UnmarshalJSON(data []byte) error {
+	type j2 JSON
+	dec := json.NewDecoder(bytes.NewReader(data))
+	return dec.Decode((*j2)(j))
```
Comment (inkel) [inline pkg/custom_models/json.go:14]: Why using a `json.Decoder` and not calling `json.Unmarshal` directly?

**Review by inkel — COMMENTED** [review]
LGTM but I've left a question before approving.

I'm wondering if it's also possible to add some tests for `*JSON.UnmarshalJSON` 🤔 especially for the use case you mention that parses numbers into `float64` rather than `json.Number`.

File: pkg/custom_models/json.go (hunk 1 above)
Comment (julienduchesne) [reply pkg/custom_models/json.go:14]: I believe the issue is that `json.Unmarshal` has the default behavior of unmarshalling as `json.Number`. I will test

**Comment by julienduchesne** [comment]
This is the real source of the issue: https://github.com/grafana/grafana-openapi-client-go/pull/26

**Pull request closed without merging**